contrib/objectstore/subscriber/secadvisor.go

```diff
@@ -25,22 +25,72 @@ import (
 	"github.com/skydive-project/skydive/flow"
 )
 
+// Version of the security advisor skydive agent
+const Version = "1.0.8"
+
 // SecurityAdvisorFlow represents a transformed flow
 type SecurityAdvisorFlow struct {
 	UUID             string
 	LayersPath       string
+	Version          string
+	Status           string
+	FinishType       string
 	Network          *flow.FlowLayer
 	Transport        *flow.TransportLayer
 	LastUpdateMetric *flow.FlowMetric
 	Metric           *flow.FlowMetric
 	Start            int64
 	Last             int64
-	FinishType       flow.FlowFinishType
+	UpdateCount      int64
 }
 
 // SecurityAdvisorFlowTransformer is a custom transformer for flows
 type SecurityAdvisorFlowTransformer struct {
-	seenFlows *cache.Cache
+	flowUpdateCount *cache.Cache
+}
+
+func (ft *SecurityAdvisorFlowTransformer) setUpadateCount(f *flow.Flow) int64 {
+	var count int64
+	if countRaw, ok := ft.flowUpdateCount.Get(f.UUID); ok {
+		count = countRaw.(int64)
+	}
+
+	if f.FinishType != flow.FlowFinishType_TIMEOUT {
+		if f.FinishType == flow.FlowFinishType_NOT_FINISHED {
+			ft.flowUpdateCount.Set(f.UUID, count+1, cache.DefaultExpiration)
+		} else {
+			ft.flowUpdateCount.Set(f.UUID, count+1, time.Minute)
+		}
+	} else {
+		ft.flowUpdateCount.Delete(f.UUID)
+	}
+
+	return count
+}
+
+func (ft *SecurityAdvisorFlowTransformer) getStatus(f *flow.Flow, updateCount int64) string {
+	if f.FinishType != flow.FlowFinishType_NOT_FINISHED {
+		return "ENDED"
+	}
+
+	if updateCount == 0 {
+		return "STARTED"
+	}
+
+	return "UPDATED"
+}
+
+func (ft *SecurityAdvisorFlowTransformer) getFinishType(f *flow.Flow) string {
+	if f.FinishType == flow.FlowFinishType_TCP_FIN {
+		return "SYN_FIN"
+	}
+	if f.FinishType == flow.FlowFinishType_TCP_RST {
+		return "SYN_RST"
+	}
+	if f.FinishType == flow.FlowFinishType_TIMEOUT {
+		return "Timeout"
```

Comment:
To be consistent should it not be in uppercase ? Ie TIMEOUT?

Reply:
This matches current implementation.
I'm not making any semantic changes.
This can be changed in the future, by decision of security advisor architects.